shared/search/docstrings.ts

```diff
@@ -13,64 +13,67 @@ import {
-        allLines[definitionLine],
-        { lineRegex, block }
-    )
-    if (sameLineDocstring) {
-        return sameLineDocstring
-    }
+    for (const {
+        lineRegex,
+        block,
+        docstringIgnore,
+        docPlacement,
+    } of commentStyles) {
+        const sameLineDocstring = findDocstringOnDefinitionLine(
+            allLines[definitionLine],
+            { lineRegex, block }
+        )
+        if (sameLineDocstring) {
+            return sameLineDocstring
+        }
 
-    if (lineRegex) {
-        const lineCommentDocstring = findDocstringInLineComments({
-            lineRegex,
-            lines: mungeLines(allLines, docPlacement, definitionLine),
-            docstringIgnore,
-        })
-        if (lineCommentDocstring) {
-            return unmungeLines(lineCommentDocstring, docPlacement).join('\n')
+        if (lineRegex) {
+            const lineCommentDocstring = findDocstringInLineComments({
+                lineRegex,
+                lines: mungeLines(allLines, docPlacement, definitionLine),
+                docstringIgnore,
+            })
+            if (lineCommentDocstring) {
+                return unmungeLines(lineCommentDocstring, docPlacement).join(
+                    '\n'
+                )
+            }
         }
-    }
 
-    if (block) {
-        // If we've reversed the lines we also need to reverse the
-        // block delimiter patterns.
-
-        const modifiedBlock =
-            docPlacement === 'below the definition'
-                ? block
-                : {
-                      startRegex: block.endRegex,
-                      endRegex: block.startRegex,
-                      lineNoiseRegex: block.lineNoiseRegex,
-                  }
-
-        const blockCommentDocstring = findDocstringInBlockComment({
-            block: modifiedBlock,
-            lines: mungeLines(allLines, docPlacement, definitionLine),
-            docstringIgnore,
-        })
-        if (blockCommentDocstring) {
-            return unmungeLines(blockCommentDocstring, docPlacement).join('\n')
+        if (block) {
+            // If we've reversed the lines we also need to reverse the
+            // block delimiter patterns.
+
+            const modifiedBlock =
+                docPlacement === 'below the definition'
+                    ? block
+                    : {
+                          startRegex: block.endRegex,
+                          endRegex: block.startRegex,
+                          lineNoiseRegex: block.lineNoiseRegex,
+                      }
+
+            const blockCommentDocstring = findDocstringInBlockComment({
+                block: modifiedBlock,
+                lines: mungeLines(allLines, docPlacement, definitionLine),
+                docstringIgnore,
+            })
+            if (blockCommentDocstring) {
+                return unmungeLines(blockCommentDocstring, docPlacement).join(
+                    '\n'
```

Comment:
Did the formatter really do this here and elsewhere?

Reply:
Yes it hits line lengths and that’s where it decided to break. Not much control over it, sometimes it’s very stupid.